lib/ansible/modules/cloud/amazon/route53.py

```diff
@@ -377,135 +382,134 @@ def commit(changes, retry_interval, wait, wait_timeout):
-        value                        = dict(required=False),
-        overwrite                    = dict(required=False, type='bool'),
-        retry_interval               = dict(required=False, default=500),
-        private_zone                 = dict(required=False, type='bool', default=False),
-        identifier                   = dict(required=False, default=None),
-        weight                       = dict(required=False, type='int'),
-        region                       = dict(required=False),
-        health_check                 = dict(required=False),
-        failover                     = dict(required=False,choices=['PRIMARY','SECONDARY']),
-        vpc_id                       = dict(required=False),
-        wait                         = dict(required=False, type='bool', default=False),
-        wait_timeout                 = dict(required=False, type='int', default=300),
-    )
-    )
-    module = AnsibleModule(argument_spec=argument_spec)
+        state=dict(aliases=['command'], choices=['present', 'absent', 'get', 'create', 'delete'], required=True),
+        zone=dict(required=True),
+        hosted_zone_id=dict(required=False, default=None),
+        record=dict(required=True),
+        ttl=dict(required=False, type='int', default=3600),
+        type=dict(choices=['A', 'CNAME', 'MX', 'AAAA', 'TXT', 'PTR', 'SRV', 'SPF', 'NS', 'SOA'], required=True),
+        alias=dict(required=False, type='bool'),
+        alias_hosted_zone_id=dict(required=False),
+        alias_evaluate_target_health=dict(required=False, type='bool', default=False),
+        value=dict(required=False, type='list'),
+        overwrite=dict(required=False, type='bool'),
+        retry_interval=dict(required=False, default=500),
+        private_zone=dict(required=False, type='bool', default=False),
+        identifier=dict(required=False, default=None),
+        weight=dict(required=False, type='int'),
+        region=dict(required=False),
+        health_check=dict(required=False),
+        failover=dict(required=False, choices=['PRIMARY', 'SECONDARY']),
+        vpc_id=dict(required=False),
+        wait=dict(required=False, type='bool', default=False),
+        wait_timeout=dict(required=False, type='int', default=300),
+    ))
+
+    # state=present, absent, create, delete THEN value is required
+    required_if = [('state', 'present', ['value']), ('state', 'create', ['value'])]
+    required_if.extend([('state', 'absent', ['value']), ('state', 'delete', ['value'])])
+
+    # If alias is True then you must specify alias_hosted_zone as well
+    required_together = [['alias', 'alias_hosted_zone_id']]
+
+    # failover, region, and weight are mutually exclusive
+    mutually_exclusive = [('failover', 'region', 'weight')]
+
+    module = AnsibleModule(argument_spec=argument_spec, required_together=required_together, required_if=required_if, mutually_exclusive=mutually_exclusive)
```

Comment:
Not really - tox.ini has max of 160 which I'm sure this is well under, I prefer 120, but basically, if I have to scroll right to read a pull request, the line is too long :).

160 is conservative, mostly for historic reasons (and there is *loads* of stuff that would fail if it were enforced), but new stuff should likely be 120. I might endeavour to add that to the development guidelines.